apis/v1alpha1/httproute_types.go

```diff
@@ -379,6 +385,35 @@ type HTTPRequestHeaderFilter struct {
 	// TODO
 }
 
+// HTTPRequestMirrorFilter defines configuration for the
+// RequestMirror filter.
+type HTTPRequestMirrorFilter struct {
```

Comment:
May be wider scoped than this PR, since the same applies to HTTPRequestHeaderFilter, but how can an implementation extend this? For example, we currently support mirroring a percentage of traffic. There are no extension points here. Would we define our own filter?

Reply:
That's a great question. I think for now the answer would have to be to define a custom filter for that use case. Although a percentage does seem straightforward enough to add here, I'm hesitant to add at this point. So far, at least for v1alpha1, we've been trying to limit fields to common functionality that is supported by most implementations. This is likely worth revisiting in the future though. Mind creating a follow up issue to track this?

Reply:
https://github.com/kubernetes-sigs/service-apis/issues/306

Reply:
I agree here.
We have to be careful about adding features in the core filters because we want most (or rather all) implementations to support it.

A meta note since we are here, we also need to figure out a way to track all custom filters that are implemented by vendors so that we can find out about convergence and react accordingly (by promoting some fields/filters into core/extended).